shop/src/test/java/ro/msg/learning/shop/LocationStrategyUnitTest.java

```diff
@@ -0,0 +1,170 @@
+package ro.msg.learning.shop;
+
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.context.annotation.Import;
+import org.springframework.security.core.parameters.P;
+import ro.msg.learning.shop.configuration.StrategyUnitTestConfig;
+import ro.msg.learning.shop.model.*;
+import ro.msg.learning.shop.repository.LocationRepository;
+import ro.msg.learning.shop.repository.StockRepository;
+import ro.msg.learning.shop.service.LocationService;
+import ro.msg.learning.shop.service.StockService;
+import ro.msg.learning.shop.service.strategy.MultipleLocationsStrategy;
+import ro.msg.learning.shop.service.strategy.SingleLocationStrategy;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.AdditionalAnswers.returnsFirstArg;
+
+@ExtendWith(MockitoExtension.class)
+public class LocationStrategyUnitTest {
+
+    @InjectMocks
+    @Qualifier("singleLocationStrategy")
+    private SingleLocationStrategy singleLocationStrategy;
+
+    @InjectMocks
+    @Qualifier("multipleLocationStrategy")
+    private MultipleLocationsStrategy multipleLocationsStrategy;
+
+    @Mock
+    private LocationService locationService;
+
+    @Mock
+    private StockService stockService;
+
+    private final List<OrderDetail> productIdAndQuantityList = new ArrayList<>();
+
+    Location location1 = new Location("location one",
```

Comment:
Don't declare your variable names with 1,2,3... if you have more than one. Give them more meaningful names.